File: CSharp Part 2/StringsAndTextProcessingHomework/Problem 12. Parse URL/ParseURL.cs
// Problem 12. Parse URL
// Write a program that parses an URL address given in the format: 
// [protocol]://[server]/[resource] and extracts from it the [protocol], [server] and [resource] elements.

using System;

class ParseURL
{
    static void Main()
    {
        Console.WriteLine("This program extracts from a given URL it's protocol, server and resource " +
                          "and prints them separately.");
        Console.Write("Please enter an URL: ");
        string url = Console.ReadLine();
        // Example = "http://telerikacademy.com/Courses/Courses/Details/212"

        int index = new int();

        index = url.IndexOf(':');
        Console.WriteLine("[protocol] = {0}", url.Substring(0, index));
        url = url.Remove(0, index + 3);

        index = url.IndexOf('/');
        Console.WriteLine(" [server]  = {0}", url.Substring(0, index));
        url = url.Remove(0, index);

        Console.WriteLine("[resource] = {0}", url);
    }
}

File: JavaScript Fundamentals/StringsHomework/Problem 7. Parse URL/ParseURL.js
var url = 'http://telerikacademy.com/Courses/Courses/Details/239';

console.log(parseURL(url));

function parseURL(url) {
    var firstSlash = url.indexOf('/'),
        secondSlash = url.indexOf('/', firstSlash + 1),
        thirdSlash = url.indexOf('/', secondSlash + 1),
            result = {};

    result.protocol = url.substring(0, firstSlash - 1);
    result.server = url.substring(secondSlash + 1, thirdSlash);
    result.resource = url.substr(thirdSlash);

    return result;
}